### PR TITLE
Topic/alire

### PR DESCRIPTION
**Pull request by godunko**

### CODE REVIEW

**Comment by Fabien-Chouteau** [comment]
Hi @godunko ,

For Alire integration, do not use such a generic external name like `BUILD_MODE`. This will likely conflict with other crates.
You should rename to `VSS_BUILD_MODE`.

https://github.com/AdaCore/VSS/blob/99a1f92d1bac839e7dbf69ed540aefd4d6293a42/gnat/vss_config.gpr#L28

**Comment by Fabien-Chouteau** [comment]
@godunko, I have a couple of other comments:

 - Is the project file https://github.com/AdaCore/VSS/blob/master/gnat/gnatcoll_text.gpr still used?
 - Why do you have two projects (`vss_text.gpr` and `vss_json.gpr`) instead of one?
 - Is it possible to move the test gprs in the testsuite directory?

**Comment by godunko** [comment]
gnatcoll_text.gpr is still in use in outside world, it will be removed at some point.

vss_text and vss_json is different independent parts of the project. It is unclear will it be possible to preserve this separation or not.

All gpr files in the same directory is just a convention, but I don't want to spread them around source tree.

**Comment by Fabien-Chouteau** [comment]
> vss_text and vss_json is different independent parts of the project. It is unclear will it be possible to preserve this separation or not.

What is the benefit of having two project files? Do you expect one to be distributed without the other?

> All gpr files in the same directory is just a convention, but I don't want to spread them around source tree.

But it pollutes the project scope.

**Comment by godunko** [comment]
> 
> What is the benefit of having two project files? Do you expect one to be distributed without the other?
> 
Yes, final idea is to have many small projects almost independent from each other. Thus, if application requires to use only text processing it may with "vss_text.gpr" only.

If this separation will be successful we will split this project into separate crates.

> > All gpr files in the same directory is just a convention, but I don't want to spread them around source tree.
> 
> But it pollutes the project scope.

Is it problem for Alire?

**Comment by Fabien-Chouteau** [comment]
> If this separation will be successful we will split this project into separate crates.

I don't think it always makes sense to split projects into multiple crates, managing crates has a cost.
And VSS is not huge.

> > But it pollutes the project scope.
> 
> Is it problem for Alire?

Not a huge problem, it is bad practice as it increase the risk of clashes.

File: gnat/vss_config.gpr [hunk 1]
```diff
@@ -25,7 +25,7 @@
 abstract project VSS_Config is
 
    type Build_Mode_Type is ("dev", "prod", "coverage");
-   Build_Mode : Build_Mode_Type := external ("BUILD_MODE", "dev");
+   Build_Mode : Build_Mode_Type := external ("VSS_BUILD_MODE", "dev");
```
Comment (Fabien-Chouteau) [inline gnat/vss_config.gpr:28]: It's better to have a default to production mode. User's of the library will most likely want best performances out of the box.

Comment (godunko) [reply gnat/vss_config.gpr:28]: I've checked with other IDE repos; "dev"/"debug" mode is default value, production environment selects "prod" mode during the builds.

Comment (Fabien-Chouteau) [reply gnat/vss_config.gpr:28]: This is a bad default because people using this library will experience the worst performances.

**Comment by reznikmm** [comment]
I've
 * rebased this PR to `master`,
 * fixed `licenses` in `alire.toml` to match the current alire convention, 
 * made `BUILD_MODE` as fallback to `VSS_BUILD_MODE`
 * chaged default for `BUILD_MODE` to `prod`

This is needed to fix #70

**Review by reznikmm — APPROVED** [review]

**Pull request merged**